### PR TITLE
refactor-new-workflows

### PR DESCRIPTION
**Pull request by jablonnc**
This PR was a significant refactor to our current approach, specifically:
- Removed `env.TEST` as a means of skipping tests within a workflow/composite action. Embedding test logic in production code is never a good idea and should be avoided. This is now being achieved via mock steps, with dynamic binding that occurs by default and can be overridden when necessary (read more about [that here](https://github.com/JupiterOne/.github/tree/refactor-new-workflows#skipping-steps)).
- Broke up a number of composite actions into more granular actions, with the end goal to make these more composable. This also entailed pulling `setup_env` out of the composite actions, and this now gets called by each job.
- Broke out all mocking logic into [mocking.ts](https://github.com/JupiterOne/.github/blob/ec34d9a1d794523e49868b5333025fa0dad4d8de/tests/utils/mocking.ts).

### CODE REVIEW

File: .github/actions/frontend/chromatic/action.yml [hunk 1]
```diff
@@ -14,36 +14,25 @@ inputs:
...
         echo "publish_chromatic=${{ fromJSON(inputs.publish_chromatic) }}"
-    - name: setup_env
-      if: ${{ !fromJSON(env.TEST) }}
-      uses: ./.github/actions/frontend/setup_env
-      with:
-        use_dev: true
-    # Upload to chromatic
     - name: chromatic_upload
-      if: ${{ !fromJSON(env.TEST) && !fromJSON(inputs.publish_chromatic) }}
+      if: ${{ !fromJSON(inputs.publish_chromatic) }}
```
Comment (Sivli-Embir) [inline .github/actions/frontend/chromatic/action.yml:27]: Should we remove these inputs and move them up to the workflow?

File: .github/actions/frontend/npm/publish/action.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      shell: bash
+      run: |
+        echo "auto_token=${{ inputs.auto_token }}"
+    - name: add_auto_to_globals
+      run: npm install -g auto@10.37.4
+      shell: bash
+    - name: deploy_to_npm
+      env:
+        GH_TOKEN: ${{ inputs.auto_token }}
+      run: npx auto shipit
```
Comment (Sivli-Embir) [inline .github/actions/frontend/npm/publish/action.yml:22]: I am wondering if we should use npx -p package mode here and skip the above install

File: .github/actions/frontend/runtime/magic_url/action.yml [hunk 3]
```diff
@@ -14,30 +14,19 @@ inputs:
...
       run: |
         echo "github_token=${{ inputs.github_token }}"
         echo "migration=${{ inputs.migration_number }}"
         echo "magic_url_route=${{ inputs.magic_url_route }}"
-    - name: setup_env
-      if: ${{ !fromJSON(env.TEST) }}
-      uses: ./.github/actions/frontend/setup_env
-      with:
-        use_dev: true
     - name: npm_build
```
Comment (Sivli-Embir) [inline .github/actions/frontend/runtime/magic_url/action.yml:26]: Should this call the build action?

**Review by Sivli-Embir — APPROVED** [review]

**Review by behobu — APPROVED** [review]

**Review by electricgull — APPROVED** [review]

File: .github/actions/frontend/npm/publish/action.yml (hunk 2 above)
Comment (jablonnc) [reply .github/actions/frontend/npm/publish/action.yml:22]: Yeah I think that's a good call, updating now in the new PR

File: .github/actions/frontend/runtime/magic_url/action.yml (hunk 3 above)
Comment (jablonnc) [reply .github/actions/frontend/runtime/magic_url/action.yml:26]: I'm actually already [doing this](https://github.com/JupiterOne/.github/blob/7f3ad83f113826f8702288a4d5083f94873b56b4/.github/workflows/frontend_runtime_application_pr.yml#L136) in the workflow, so this is unnecessary. Good catch!

**Pull request merged**